src/ledger/backend/contracts_deposit.spec.ts

```diff
@@ -4,38 +4,58 @@
 import { expect } from "chai";
 
 import { Wallet } from "ethers";
-import { ETHZERO, Amount } from "#erdstall/ledger/assets";
+import { ETHZERO, Amount, Asset } from "#erdstall/ledger/assets";
 import { Address } from "#erdstall/ledger";
 import {
 	makeETHDepositCalls,
 	makeERC20DepositCalls,
 	makeERC721DepositCalls,
 } from "./contracts_deposit";
+import { DepositerCallsFactory } from "./tokenmanager";
+import { PerunArt__factory } from "./contracts/factories/PerunArt__factory";
 
 import { Enviroment, setupEnv } from "#erdstall/test/ledger";
+import * as test from "#erdstall/test";
+
+const TOKEN_SIZE = 4;
 
 describe("Deposit_Call_Wrapper", () => {
+	const rng = test.newPrng();
 	let testenv: Enviroment;
 	let bob: Wallet;
-	const amount = new Amount(100n);
+	const amount = new Amount(test.newRandomBigInt(rng, 16));
+	const tokens = test.newRandomTokens(rng, TOKEN_SIZE);
 
 	before(async () => {
 		testenv = await setupEnv();
 		bob = testenv.users[0];
+
+		const part = PerunArt__factory.connect(testenv.perunArt, bob);
+		for (const id of tokens.value) {
+			await part.mint(bob.address, id);
```

Comment:
So `bob` is configured as being a minter before? Otherwise this `mint` call will revert.